Lib/argparse.py

```diff
@@ -704,6 +704,8 @@ def _get_action_name(argument):
         return argument.metavar
     elif argument.dest not in (None, SUPPRESS):
         return argument.dest
+    elif argument.choices:
+        return '{' + ','.join(argument.choices) + '}'
```

Comment:
The test doesn’t really help me understand what this change does (i.e. what a human would see in an error case).

I’m also not sure if some of the changes in the various patches for bpo-9253 (especially in tests) could be useful here.

Reply:
I'll check those out. I had a difficult time demonstrating (by assertion) the actual output (didn't find any examples) -- I'll bang against that and hopefully I'll make something more expressive :)

Reply:
@merwok looks like they had the same idea I had (but a different implementation) -- I took one of their tests for additional coverage and updated the tests to hopefully be more expressive (and have a better failure mode)

Reply:
That’s great.  Please name the other authors in NEWS!

Reply:
Code for this is already in `_metavar_formatter`:
```python
        elif action.choices is not None:
            choice_strs = [str(choice) for choice in action.choices]
            result = '{%s}' % ','.join(choice_strs)
```

The metavar formatter can be called, if the parser is known (as `self`), using:
```python
            default = self._get_default_metavar_for_positional(action)
            metavar, = self._metavar_formatter(action, default)(1)
```
Perhaps `_get_action_name` should be made into a method, or take the parser as an optional argument?